testsuite/test_config_users.py

```diff
@@ -0,0 +1,209 @@
+import pytest
+from rest_framework.test import APIClient
+from django.contrib.auth.models import User
+
+pytestmark = pytest.mark.django_db
+
+@pytest.fixture
+def client():
+    return APIClient()
+
+@pytest.fixture
+def admin_user():
+    return User.objects.create_user(
+        username='admin',
+        password='q',
+        is_staff=True,
+    )
+
+def test_get_users(client, admin_user):
+    assert client.login(username='admin', password='q')
+
+    users_resp = client.get('/api/users/')
+    assert users_resp.status_code == 200
+    assert [u['username'] for u in users_resp.json()] == ['admin']
+
+def test_create_users(client, admin_user):
+    assert client.login(username='admin', password='q')
+
+    create = client.post('/api/users/', data={
+        "username": "mike",
+        "is_admin": False,
+        "first_name": "Mike",
+        "last_name": "Tyson",
+        "is_active": True,
+    })
+    assert create.status_code == 201
+
+    user_list = client.get("/api/users/")
+    assert set(u['username'] for u in user_list.json()) == set(["admin", "mike"])
+
+    get_mike = client.get('/api/users/mike/')
+    assert get_mike.status_code == 200
+    assert "mike" == get_mike.json()['username']
+    assert False == get_mike.json()['is_admin']
+    assert "Mike" == get_mike.json()['first_name']
+    assert "Tyson" == get_mike.json()['last_name']
+    assert True == get_mike.json()['is_active']
+
+def test_password_change_admin_self(client, admin_user):
+    assert client.login(username='admin', password='q')
+
+    # test invalidation of short password
+    password = client.post("/api/users/admin/password/", data={
+        "new_password": "qqq",
+    })
+    assert password.status_code == 400
+    assert password.json() == {"new_password": ["password too short"]}
+
+    # make sure old password still works
+    client.logout()
+    assert client.login(username='admin', password='q')
+
+    # change the password
+    password = client.post("/api/users/admin/password/", data={
+        "new_password": "12345678",
+    })
+    assert password.status_code == 200
+
+    # make sure new login works
+    client.logout()
+    client.login(username='admin', password='12345678')
+
+def test_password_change_nonadmin(client):
```

Comment:
Ok, this test is super useful, but do we have a test for when a user tries to change somebody else's password?